src/input/mouse.c

```diff
@@ -298,8 +298,19 @@ void Mouse_HandleMovement(uint16 newButtonState, uint16 mouseX, uint16 mouseY)
 {
 	g_mouseLock = 0x1;
 
+#if defined(_WIN32) && defined(WITH_SDL2)
+	if (g_scale_filter == FILTER_NEAREST_NEIGHBOR) {
+		g_mouseX = mouseX / g_screen_magnification;
+		g_mouseY = mouseY / g_screen_magnification;
```

Comment:
It's necessary since SDL_RenderSetLogicalSize is no more used: it seems can't manage very well cursor pointer x/y limits.
Alternatively SDL_RenderCopy is managed by Dst Rect.